app/models/mixins/miq_policy_mixin.rb

```diff
@@ -109,7 +109,15 @@ def check_policy_prevent_callback(*action, _status, _message, result)
       data  = event.attributes["full_data"]
       prevented = data.fetch_path(:policy, :prevented) if data
     end
-    prevented ? _log.info(event.attributes["message"]) : send(*action)
+
+    if prevented
+      _log.info(event.attributes["message"])
+    elsif action.last.kind_of?(Hash)
+      hsh = action.pop
+      send(*action, **hsh)
+    else
+      send(*action)
+    end
```

Comment:
There's already an activesupport method that simplifies this pattern, so perhaps:

```suggestion
    else
      kwargs = action.extract_options!
      send(*action, **kwargs)
    end
```

Reply:
No, there's no need to do that.  If there are no kwargs passed then extract_options! returns an empty hash., and splatting an empty hash works fine with kwargs.

For example:

Ruby 2.7:

```
02:12:52 ~/dev/intersight-sdk-ruby (main *) (2.7.4) ✓ irb
[1] pry(main)> def show(param1, paramArg: "default"); puts "param1: #{param1}, paramArg: #{paramArg}"; end
[2] pry(main)> show("a")
param1: a, paramArg: default
[3] pry(main)> show("a", **{})
param1: a, paramArg: default
[4] pry(main)> show("a", paramArg: "b")
param1: a, paramArg: b
[5] pry(main)> show("a", **{paramArg: "b"})
param1: a, paramArg: b
```

Ruby 3.0

```
irb(main):001:0> def show(param1, paramArg: "default"); puts "param1: #{param1}, paramArg: #{paramArg}"; end
irb(main):002:0> show("a")
param1: a, paramArg: default
irb(main):003:0> show("a", **{})
param1: a, paramArg: default
irb(main):004:0> show("a", paramArg: "b")
param1: a, paramArg: b
irb(main):005:0> show("a", **{paramArg: "b"})
param1: a, paramArg: b
```

Here is what extract_options does:

```
[1] pry(main)> require "active_support/all"
[2] pry(main)> ["a", "b", {paramArg: "c"}].extract_options!
=> {:paramArg=>"c"}
[3] pry(main)> ["a", "b", {}].extract_options!
=> {}
[4] pry(main)> ["a", "b"].extract_options!
=> {}
```